### PR TITLE
feat(services): revamp service details sidebar

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-768

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery  preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery  preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Make it work, make it right, make it fast

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] staging - feat(services): revamp service details sidebar - 2023-09-28T15:26:51Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/5a2fc823-b1af-4736-8828-517322f96007/services/general)

 Another comment will be posted when deployments are terminated

File: libs/domains/services/feature/src/lib/last-commit/last-commit.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import { useCommits } from '../hooks/use-commits/use-commits'
+
+export interface LastCommitProps {
+  className?: string
+  gitRepository: ApplicationGitRepository
+  serviceId: string
+  serviceType: Extract<ServiceType, 'APPLICATION' | 'JOB' | 'CRON_JOB' | 'LIFECYCLE_JOB'>
+}
+
+export function LastCommit({ className, gitRepository, serviceId, serviceType }: LastCommitProps) {
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/last-commit/last-commit.tsx:16]: The behavior is a little bit glitchy when you hover the component and change the icon, the width is different, should be similar to be more aesthetic. Moreover, we have a gray border on the "number of commits", it should be removed or orange

https://www.loom.com/share/156479caeef24448bf78f8bf8fea8c32

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                      <Dt>Repository:</Dt>
+                      <Dd>
+                        <a href={gitRepository.url} target="_blank" rel="noopener noreferrer">
+                          <Badge variant="surface" size="xs" className="gap-2">
+                            <Icon
+                              name={gitRepository.url.includes('//github') ? IconEnum.GITHUB : IconEnum.GITLAB}
+                              height={14}
+                              width={14}
+                            />
+                            <Truncate text={gitRepository.name} truncateLimit={20} />
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/service-details/service-details.tsx:145]: The repository "tags" overflows a little onto the right padding

Capture d’écran 2023-10-02 à 11 38 33

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+      {resources && (
+        <>
+          <hr />
+          <Section className={sectionClassName}>
+            <div className="flex flex-row justify-between">
+              {/* XXX: Should be Heading, typography & design wanted */}
+              <span className="flex items-center gap-1 text-neutral-400 font-medium">
+                Resources
+                <Tooltip content="This is based on the configuration of your service">
+                  <span className="flex items-center">
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/service-details/service-details.tsx:204]: Ugly but visually better

```suggestion
                  <span className="flex items-center relative top-[1px]">
```

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx [hunk 4]
```diff
@@ -100,7 +99,8 @@ export function ServiceDetails({ className, environmentId, serviceId, ...props }
   return (
     <div className={twMerge('flex flex-col shrink-0 gap-5 py-8 text-sm', className)} {...props}>
       <Section className={sectionClassName}>
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/service-details/service-details.tsx:101]: The bottom margin for each section isn't good, should be `20px`
Capture d’écran 2023-10-02 à 11 34 34

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    }))
+    .otherwise(() => undefined)
+
+  const ResourceUnit = ({ value, description }: { value?: string | number | null; description: string }) => {
+    if (value === undefined || value === null) {
+      return
+    }
+    return (
+      <div className="grid grid-row-2">
+        <span className="text-lg text-neutral-400 font-semibold">{value}</span>
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/service-details/service-details.tsx:57]: Should be 16px

```suggestion
        <span className="text-base text-neutral-400 font-semibold">{value}</span>
```

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        <span>{description}</span>
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/service-details/service-details.tsx:58]: Should be 14px
```suggestion
        <span className="text-sm">{description}</span>
```

**Review by RemiBonnet — CHANGES_REQUESTED** [review]

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx (hunk 5 above)
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/service-details/service-details.tsx]: https://github.com/Qovery/console/pull/865/commits/afcf4d0338229cdfeff146f720bbdb191ee9393d

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx (hunk 3 above)
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/service-details/service-details.tsx:204]: This is not necessary in the latest version

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/service-details/service-details.tsx]: I think you haven't test the latest version.

File: libs/domains/services/feature/src/lib/last-commit/last-commit.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/last-commit/last-commit.tsx:16]: > The behavior is a little bit glitchy when you hover the component and change the icon, the width is different, should be similar to be more aesthetic.
Fixed
https://github.com/Qovery/console/pull/865/commits/73cc81edd85353abea293a1ad00e0d4a51e34b76

> Moreover, we have a gray border on the "number of commits", it should be removed or orange
Fixed https://github.com/Qovery/console/pull/865/commits/7e1c502f0339cce7b5ba83a7fb15846ecea14449
Still ugly code unless we truly think about the inset component

File: libs/shared/util-js/src/lib/format-bytes.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import prettyBytes from 'pretty-bytes'
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/shared/util-js/src/lib/format-bytes.ts:2]: https://github.com/Qovery/console/pull/865/commits/22263174351967f54500ef7668fcdcdef6c2437c

File: libs/shared/util-js/src/lib/format-cron-expression.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import cronstrue from 'cronstrue'
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/shared/util-js/src/lib/format-cron-expression.ts:2]: https://github.com/Qovery/console/pull/865/commits/22263174351967f54500ef7668fcdcdef6c2437c

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+            <div className="flex flex-row justify-between">
+              <Heading className="flex items-center gap-1">
+                Resources
+                <Tooltip content="This is based on the configuration of your service">
+                  <span className="flex items-center">
+                    <Icon name={IconAwesomeEnum.CIRCLE_INFO} className="text-xs text-neutral-350" />
+                  </span>
+                </Tooltip>
+              </Heading>
+              <Link to="../settings/resources" relative="path">
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/service-details/service-details.tsx:209]: I don't think this should be changed. This is a link not button from a semantic html point of view.
Using `ButtonLegacy` will resolve into `Link` anyway and this is a component we don't want to use anymore.

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx (hunk 4 above)
Comment (RemiBonnet) [reply libs/domains/services/feature/src/lib/service-details/service-details.tsx:101]: Ha yes it's better!

File: libs/domains/services/feature/src/lib/last-commit/last-commit.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/domains/services/feature/src/lib/last-commit/last-commit.tsx:16]: Nice thanks !

Just one thing, when we reload the page, the commit component appears after the content: https://www.loom.com/share/9a6145aabac74ffb9bd7f623b21bfdac

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/domains/services/feature/src/lib/service-details/service-details.tsx:215]: ok but it'sbetter if you use the new Link component, no? And if you can add a hover color with a transition should be perfect 💯

File: libs/domains/services/feature/src/lib/last-commit/last-commit.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+      setCopied(false)
+    }, 1000)
+  }
+
+  if (deployedCommit === undefined) {
+    return null
+  }
+
+  return (
+    <div className="inline-flex gap-2">
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/last-commit/last-commit.tsx:37]:
```suggestion
    <div className="inline-flex items-center gap-2">
```

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/service-details/service-details.tsx]: https://github.com/Qovery/console/pull/865/commits/ada4230d54dd4f479c8ca428bfbc6d951962ef96

**Review by RemiBonnet — APPROVED** [review]

File: libs/domains/services/feature/src/lib/service-details/service-details.tsx [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+import { useService } from '../hooks/use-service/use-service'
+import { LastCommit } from '../last-commit/last-commit'
+import { ServiceDetailsSkeleton } from './service-details-skeleton'
+
+export interface ServiceDetailsProps extends ComponentPropsWithoutRef<'div'> {
+  environmentId: string
+  serviceId: string
+}
+
+export function ServiceDetails({ className, environmentId, serviceId, ...props }: ServiceDetailsProps) {
```
Comment (RemiBonnet) [inline libs/domains/services/feature/src/lib/service-details/service-details.tsx:27]: Not sure about this name, but maybe `ServiceSidebarDetails` or something like that?

I think `ServicesDetails` is too general

Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/service-details/service-details.tsx:27]: Ahah I think the opposite :P
UI component and Product component are 2 different things and should not be tied together.
This component name is general on purpose so we can put it in a sidebar, in main content page or whatever.
What matter is what functionality it brings from a product perspective not the container.

Comment (RemiBonnet) [reply libs/domains/services/feature/src/lib/service-details/service-details.tsx:27]: Yes I know this point. If tomorrow we have other service details, not in the same place, but before the metric, for example, we will have to rename this component, no?

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**